server/src/main/java/org/opensearch/repositories/blobstore/AbstractBlobStoreFormat.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.blobstore;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.store.OutputStreamIndexOutput;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.lucene.store.IndexOutputOutputStream;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.core.xcontent.MediaTypeRegistry;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Provides common methods, variables that can be used by the implementors.
+ *
+ * @opensearch.internal
+ */
+public class AbstractBlobStoreFormat<T extends ToXContent> {
```

Comment:
should we rename it to BaseBlobStoreFormat as Abstract can be misleading without any abstract method

Reply:
Addressed.